data/shared/src/main/scala/sigma/ast/values.scala

```diff
@@ -210,6 +210,30 @@ object Value {
     c > 0
   }
 
+  /** Traverses the given expression tree and counts the number of operations
+   * which read the blockchain context. If it is non-zero, returns true. */
+  def isUsingBlockchainContext(exp: SValue): Boolean = {
+    val blockchainContextNode: PartialFunction[Any, Int] = {
+      case Height => 1
+      case LastBlockUtxoRootHash => 1
+      case MinerPubkey => 1
+      case MethodCall(obj, method, _, _) =>
+        (obj, method.objType) match {
```

Comment:
Here, it seems, no need to match on tuple.

Reply:
Updated in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/929/commits/22f058625a15767926d53b23ce5812a0e78a7845.